### DISCUSSION TITLE
Authentication, Authorization and Accounting (AAA) management

### DISCUSSION THREAD

**Discussion by firozemzr** [OSBDAlliance/headlesscms#7] [Ideas]
CMS should provide extensive AAA control for content management and delivery - specially for authenticated and un-authenticated users, authorization management for content management, delivery and other function. 

Also extensive logging/accounting of content consumption across different device, user-agents, network, users including content billing/charging (prepaid and postpaid) so that detail analytics can be built on this accounting data.

**Comment by mohammad-arif-rahim** [comment]
It should provide **Token expiration** and **token rotation** for cross platform authentication. It should also support **OAuth 2.0** and **JWT**.  I dont think it covers full AAA scope and to cover full AAA scope is only necessary for security management products.

**Reply by firozemzr** [reply]
Covering jwt anf oauth 2.0 or tokenization is a good idea.

But it has to cover quite detail AAA specially for news agencies and different other content business models where pay per use or quota based content subscriptions are needed.

For example, news agencies may have no of content subscription or no of time content access/throughput based class of service for different content subscribers.

Without proper AAA these billings and access restrictions will not be possible.
